python/dask_cudf/dask_cudf/core.py

```diff
@@ -90,11 +92,30 @@ def __init__(self, dsk, name, meta, divisions):
         self._meta = meta
         self.divisions = tuple(divisions)
 
-    def __getstate__(self):
-        return (self.dask, self._name, self._meta, self.divisions)
-
-    def __setstate__(self, state):
-        self.dask, self._name, self._meta, self.divisions = state
+    def serialize(self):
+        meta_header, meta_frames = self._meta.serialize()
+        header = {
+            "dask": pickle.dumps(self.dask),
```

Comment:
We might be able to get away with not pickling this and just adding it to the `header` as-is. Though am a little worried  that issues like ( https://github.com/dask/distributed/issues/3716 ) might bite us. Though that bug seems solvable now. I suppose we could just try it and fix any bugs that crop up. WDYT?

Reply:
Seems reasonable to try - So, is the header typically serialized with msgpack? Isn't it possible for the user to inject some messy types into the graphmsgpack will choke on? (sorry for my lack of knowledge here)

Reply:
Yeah I think the header is required to be msgpack serializable.

Good point! I suppose we can call `serialize(...)` on the graph. Or are there other pitfalls you imagine here?